src/backend/session.rs

```diff
@@ -0,0 +1,86 @@
+use std::{fmt::Debug, sync::Arc};
+
+use async_trait::async_trait;
+use zbus::{
+    object_server::SignalEmitter,
+    zvariant::{ObjectPath, OwnedObjectPath, SerializeDict, Type},
+};
+
+use crate::desktop::HandleToken;
+
+pub(crate) struct Session {
+    path: OwnedObjectPath,
+    monitor: Option<Arc<dyn SessionMonitor>>,
+}
+
+impl Session {
+    #[allow(dead_code)]
+    pub fn path(&self) -> ObjectPath<'_> {
+        self.path.as_ref()
+    }
+
+    pub(crate) fn new(path: OwnedObjectPath, monitor: Option<Arc<dyn SessionMonitor>>) -> Self {
+        Self { path, monitor }
+    }
+
+    pub async fn serve(self, cnx: zbus::Connection) -> zbus::Result<bool> {
+        cnx.object_server().at(self.path.clone(), self).await
+    }
+}
+
+#[zbus::interface(
+    name = "org.freedesktop.impl.portal.Session",
+    proxy(gen_blocking = false)
+)]
+impl Session {
+    #[zbus(property(emits_changed_signal = "const"), name = "version")]
+    fn version(&self) -> u32 {
+        1
+    }
+
+    async fn close(
+        &self,
+        #[zbus(object_server)] server: &zbus::ObjectServer,
+    ) -> zbus::fdo::Result<()> {
+        #[cfg(feature = "tracing")]
+        tracing::debug!("Closing session {:?}", self.path.as_str());
```

Comment:
do you need {:?} when using as_str instead of just {}?

Reply:
No, indeed.